### PR TITLE
SONARJAVA-372 Handle constants and annotated fields

### PR DESCRIPTION
**Pull request by m-g-sonar**

### CODE REVIEW

File: java-checks/src/main/java/org/sonar/java/checks/PrivateFieldUsedLocallyCheck.java [hunk 1]
```diff
@@ -61,11 +63,17 @@ public void visitNode(Tree tree) {
...
     classSymbol.memberSymbols().stream()
       .filter(PrivateFieldUsedLocallyCheck::isPrivateField)
+      .filter(s -> !(s.isFinal() && s.isStatic()))
+      .filter(s -> !useAuthorizedAnnotation(s))
       .filter(s -> !s.usages().isEmpty())
       .filter(s -> !fieldsReadOnAnotherInstance.contains(s))
       .forEach(s -> checkPrivateField(s, classSymbol));
   }
 
+  private static boolean useAuthorizedAnnotation(Symbol s) {
```
Comment (tvsonar) [inline java-checks/src/main/java/org/sonar/java/checks/PrivateFieldUsedLocallyCheck.java]: `hasAuthorizedAnnotation` or `useAllowedAnnotation` sounds better to me.

File: java-checks/pom.xml [hunk 2]
```diff
@@ -184,11 +184,6 @@
                   <type>jar</type>
                 </artifactItem>
                 <artifactItem>
-                  <groupId>javax</groupId>
```
Comment (tvsonar) [inline java-checks/pom.xml]: Is this change required?

Comment (m-g-sonar) [inline java-checks/pom.xml]: It's a duplicate of line [#139](https://github.com/SonarSource/sonar-java/pull/943/files/9791754d17d30557a03aa303e233cda221181131#diff-598a04ed7e994c371420982332fc73caR139), that I discovered while I wanted to add the jar containing `@Inject` annotation.
